### PR TITLE
fix: fix long initial time with huge print history

### LINKED ISSUE #1468: Slow website UI load times
### Mainsail Version:

v2.5.1 _(Edit: Just updated to v2.6.0 and still have the same issue)_

### Browser:

Chrome

### Device:

Desktop PC

### Operating System:

Windows

### What happened?

When I load the webpage it takes around 8-20 seconds to load. (Validated on multiple high-end devices). 

### What did you expect to happen instead?

The page to load a lot faster than it currently does

### How to reproduce this bug?

If you have a lot of history file, you will see this slow down. I just with a lot of use time (438hours) 

### Additional information:

hso the issue seems to be that it loads the entire history on page I it. My suggestion is to have it load the top X from history (maybe 10?) and the rest of the history can be loaded on the history page load 

image

image

https://github.com/Rat-OS/RatOS-configuration/assets/4393979/61c7380c-8376-4ade-ab3c-a7b693c39aa6

https://github.com/Rat-OS/RatOS-configuration/assets/4393979/4aa5a27d-b621-418f-bfd6-3ccf810838ad

You can find my DB here: 
[data.zip](https://github.com/mainsail-crew/mainsail/files/12073713/data.zip)

### PR DESCRIPTION
**Pull request by meteyou**
## Description

This PR fix a issue with long initial times with huge print histories. It only loads the first 100 entries and the rest only when the user clicks on a button.

## Related Tickets & Documents

fixes #1468 

## Mobile & Desktop Screenshots/Recordings

none

## [optional] Are there any post-deployment tasks we need to perform?

none

### CODE REVIEW

File: src/components/panels/HistoryListPanel.vue [hunk 1]
```diff
@@ -523,14 +518,15 @@ import {
     mdiNotebookEdit,
     mdiNotebookPlus,
     mdiNotebook,
+    mdiDownload,
```
Comment (rackrick) [inline src/components/panels/HistoryListPanel.vue:521]: remove unused icon reference

File: src/locales/de.json [hunk 2]
```diff
@@ -351,6 +351,7 @@
         "Jobs": "Drucke",
         "LastModified": "Zuletzt geändert",
         "LayerHeight": "Schichthöhe",
+        "LoadCompleteHistory": "Lade komplette Verlaufsliste",
```
Comment (rackrick) [inline src/locales/de.json:354]: We use `Historie` on different keys why the switch to `Verlaufsliste`?

```suggestion
        "LoadCompleteHistory": "Lade vollständige Historie",
```

File: src/locales/en.json [hunk 3]
```diff
@@ -351,6 +351,7 @@
         "Jobs": "Jobs",
         "LastModified": "Last Modified",
         "LayerHeight": "Layer Height",
+        "LoadCompleteHistory": "Load complete history jobs",
```
Comment (rackrick) [inline src/locales/en.json:354]:
```suggestion
        "LoadCompleteHistory": "Load complete history",
```

**Review by rackrick — APPROVED** [review]

**Comment by jLynx** [comment]
Thanks for completing this, excited to try it out. Happy 2024!

**Comment by jLynx** [comment]
Issue seems to be mostly solved now. Now it takes no longer than 8 seconds which is a massive improvement over 15-30 seconds previously. Looking at the natwork traffic, 2 seconds is for the css file. So I'm assuming thst gets dynamically generated?

**Comment by meteyou** [comment]
@jLynx no... All files are static.

**Pull request merged**